src/test/kotlin/com/igorwojda/list/product/challenge.kt

```diff
@@ -4,7 +4,12 @@ import org.amshove.kluent.shouldBeEqualTo
 import org.junit.jupiter.api.Test
 
 private fun product(list: List<Int>): Int {
-    TODO("not implemented")
+    // Implementation below does the same thing, but with less code
+    //    return list.reduce { it, product -> product * it }
+
+    // But since you asked to have all the required concepts, please check the implementation below :)
+    operator fun List<Int>.invoke() = reduce { it, product -> it * product }
```

Comment:
Why is this line required?

Reply:
Yes, you are right. It is not required, I can implement this challenge with only the following line:

`return list.reduce(Int::times)`